pkg/proc/variables_test.go

```diff
@@ -2053,3 +2053,72 @@ func TestCapturedVarVisibleOnFirstLine(t *testing.T) {
 		}
 	})
 }
+
+func TestIssue4116(t *testing.T) {
+	protest.MustSupportFunctionCalls(t, testBackend)
+	protest.AllowRecording(t)
+
+	varTestcases := []varTest{
+		{"v.Model", true, "\"B\"", "", "string", nil},
+		{"v.A.Model", false, "main.(*A).Model", "", "func() string", nil},
+		{"v1.X", false, "main.(*B1).X", "", "func()", nil},
+		{"v1.A1.X", false, "0", "", "int", nil},
+		{"v2.X", false, "main.(*B2).X", "", "func()", nil},
+		{"v2.B2.X", false, "main.(*B2).X", "", "func()", nil},
+		{"v2.B2.A2.X", true, "0", "", "int", nil},
+		{"v2.A2.X", true, "0", "", "int", nil},
+		{"v7.X", false, "main.A7.X", "", "func()", nil},
+		{"x.X", false, "main.(*A7).X", "", "func()", nil},
+		{"x1.X", false, "", "", "", errors.New("x1 (type void) is not a struct")},
+		{"x2.X", false, "main.(*A7).X", "", "func()", nil},
+		{"x3.X", false, "main.(*A7).X", "", "func()", nil},
+		{"v9.V.X", false, "main.(*B9).X", "", "func()", nil},
+		{"a.X", false, "main.A8.X", "", "func()", nil},
+		{"b.X", false, "", "", "", errors.New("b has no member X")},
+		{"c.X", false, "main.(*A8).X", "", "func()", nil},
+		{"d.X", true, "1", "", "int", nil},
+	}
+	funcTestcases := []testCaseCallFunction{
+		{"v.A.Model()", []string{":string:\"A\""}, nil, 0},
+		{"v1.X()", []string{}, nil, 0},
+		{"v2.X()", []string{}, nil, 0},
+		{"v2.B2.X()", []string{}, nil, 0},
+		{"v3.X()", []string{}, nil, 0},
+		{"v3.B3.X()", []string{}, nil, 0},
+		{"v4.X()", []string{}, nil, 0},
+		{"v4.TestX.X()", []string{}, nil, 0},
+		{"v5.X", []string{":int:0"}, nil, 0},
+		{"v5.B5.X()", []string{":string:\"anonymous func\""}, nil, 0},
+		{"v5.B5.A5.X()", []string{":string:\"anonymous func\""}, nil, 0},
+		{"v5.A5.X()", []string{":string:\"anonymous func\""}, nil, 0},
+		{"v5.A5.X()", []string{":string:\"anonymous func\""}, nil, 0},
+		{"*(ch.buf)", []string{":[0]int:[0]int []"}, nil, 0},
+		{"*(v6.Chan.buf)", []string{":[0]struct struct {}:[0]struct struct {} []"}, nil, 0},
+		{"*(v6.buf)", []string{}, errors.New("v6 has no member buf"), 0},
+	}
+
+	withTestProcess("issue4116", t, func(p *proc.Target, grp *proc.TargetGroup, fixture protest.Fixture) {
+		testCallFunctionSetBreakpoint(t, p, grp, fixture)
+
+		assertNoError(grp.Continue(), t, "Continue()")
+
+		for _, tc := range varTestcases {
+			variable, err := evalVariableWithCfg(p, tc.name, pnormalLoadConfig)
+			if tc.err == nil {
+				assertNoError(err, t, "EvalVariable() returned an error")
+				assertVariable(t, variable, tc)
+			} else {
+				if err == nil {
+					t.Fatalf("Expected error %s, got no error: %s\n", tc.err.Error(), api.ConvertVar(variable).SinglelineString())
+				}
+				if tc.err.Error() != err.Error() {
+					t.Fatalf("Unexpected error. Expected %s got %s", tc.err.Error(), err.Error())
+				}
+			}
+		}
+
+		for _, tc := range funcTestcases {
+			testCallFunction(t, grp, p, tc)
```

Comment:
This is too much, just check that the returned variable has the correct name, calling them is slow and complicates things.